cli/command/stack/kubernetes/stack.go

```diff
@@ -0,0 +1,106 @@
+package kubernetes
+
+import (
+	"io/ioutil"
+	"path"
+	"sort"
+
+	"github.com/docker/cli/kubernetes/compose/v1beta2"
+	"github.com/docker/cli/kubernetes/labels"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// stack is the main type used by stack commands so they remain independent from kubernetes compose component version.
+type stack struct {
+	name        string
+	composeFile string
+	spec        *v1beta2.StackSpec
+}
+
+// getServices returns all the stack service names, sorted lexicographically
+func (s *stack) getServices() []string {
+	services := make([]string, len(s.spec.Services))
+	for i, service := range s.spec.Services {
+		services[i] = service.Name
+	}
+	sort.Strings(services)
+	return services
+}
+
+// createFileBasedConfigMaps creates a Kubernetes ConfigMap for each Compose global file-based config.
+func (s *stack) createFileBasedConfigMaps(configMaps corev1.ConfigMapInterface) error {
+	for name, config := range s.spec.Configs {
+		if config.File == "" {
+			continue
+		}
+
+		fileName := path.Base(config.File)
+		content, err := ioutil.ReadFile(config.File)
+		if err != nil {
+			return err
+		}
+
+		if _, err := configMaps.Create(toConfigMap(s.name, name, fileName, content)); err != nil {
```

Comment:
Is "create" idempotent, or should we check if a configmap already exists (and is unmodified - if we want to keep the immutable behavior)

Reply:
Create will fail if it already exists, so I think it's ok?